contracts/FiatToken.cdc

```diff
@@ -1600,100 +1146,152 @@ pub contract FiatToken: FungibleToken {
+        ownerAccount.link<&Owner{OwnerCapReceiver}>(self.OwnerCapReceiverPubPath, target: self.OwnerStoragePath)
 
-        // Set up the Pauser
-        //
-         adminAccount.save(<-ownerExecutorRef.createNewPauseExecutor(), to: self.PauseExecutorStoragePath)
+        // Setup the MasterMinter
+        pubKeyAttrs = []
+        i = 0
+        while i < masterMinterPubKeys.length {
+            let pka = OnChainMultiSig.PubKeyAttr(sa: masterMinterPubKeysAlgos[i], w: masterMinterPubKeysWeights[i])
+            pubKeyAttrs.append(pka)
+            let key = PublicKey(
+                publicKey: masterMinterPubKeys[i].decodeHex(), 
+                signatureAlgorithm: SignatureAlgorithm(rawValue: masterMinterPubKeysAlgos[i]) ?? panic ("Invalid signature algo")
+            )
+            masterMinterAccount.keys.add(
+                publicKey: key,
+                hashAlgorithm: HashAlgorithm.SHA3_256,
+                weight: masterMinterPubKeysWeights[i]
+            )
+            i = i + 1
+        }
+        masterMinterAccount.save(<- create MasterMinter(pk: masterMinterPubKeys, pka: pubKeyAttrs), to: self.MasterMinterStoragePath)
+        masterMinterAccount.link<&MasterMinter{OnChainMultiSig.PublicSigner}>(self.MasterMinterPubSigner, target: self.MasterMinterStoragePath)
+        masterMinterAccount.link<&MasterMinter{ResourceId}>(self.MasterMinterUUIDPubPath, target: self.MasterMinterStoragePath)
+        masterMinterAccount.link<&MasterMinter{MasterMinterCapReceiver}>(self.MasterMinterCapReceiverPubPath, target: self.MasterMinterStoragePath)
 
-        let pauser <- FiatToken.createNewPauser(publicKeys: ownerAccountPubKeys, pubKeyAttrs: pubKeyAttrs)
-        adminAccount.save(<-pauser, to: self.PauserStoragePath)
-        adminAccount.link<&Pauser{OnChainMultiSig.PublicSigner}>(self.PauserPubSigner, target: self.PauserStoragePath)
-        //adminAccount.link<&Pauser{ResourceId}>(self.PauserUUIDPubPath, target: self.PauserStoragePath)
-        adminAccount.link<&Pauser{PauseCapReceiver}>(self.PauserCapReceiverPubPath, target: self.PauserStoragePath)
+        // Setup the Blocklister 
+        pubKeyAttrs = []
+        i = 0
+        while i < blocklisterPubKeys.length {
+            let pka = OnChainMultiSig.PubKeyAttr(sa: blocklisterPubKeysAlgos[i], w: blocklisterPubKeysWeights[i])
+            pubKeyAttrs.append(pka)
+            let key = PublicKey(
+                publicKey: blocklisterPubKeys[i].decodeHex(), 
+                signatureAlgorithm: SignatureAlgorithm(rawValue: blocklisterPubKeysAlgos[i]) ?? panic ("Invalid signature algo")
+            )
+            blocklisterAccount.keys.add(
+                publicKey: key,
+                hashAlgorithm: HashAlgorithm.SHA3_256,
+                weight: blocklisterPubKeysWeights[i]
+            )
+            i = i + 1
+        }
+        blocklisterAccount.save(<- create Blocklister(pk: masterMinterPubKeys, pka: pubKeyAttrs), to: self.BlocklisterStoragePath)
```

Comment:
`pk: masterMinterPubKeys`

Should be `blocklisterPubKeys` .

Reply:
good catch, fixed!

Reply:
Awesome!